components/atoms/ItemizationBox/ItemizationBox.js

```diff
@@ -0,0 +1,201 @@
+/** @module ItemizationBox */
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, {createRef, PureComponent} from 'react';
+import {animated, config, Spring} from 'react-spring/renderprops.cjs';
+import Button from '~components/atoms/Button/Button';
+import Tooltip from '~components/atoms/Tooltip/Tooltip';
+import {formatCurrency} from '~components/utilities/FormatUtils/FormatUtils';
+import './ItemizationBox.scss';
+
+/** Displays a collapsible  box which itemizes and sums totals.  */
+class ItemizationBox extends PureComponent {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      height: props.isCollapsed ? 0 : 'auto',
+    };
+
+    this.contentNode = createRef();
+  }
+
+  /** @inheritdoc */
+  componentDidUpdate(prevProps) {
+    const updatedState = {};
+
+    // Toggles the height if the component was previously collapsed.
+    if (prevProps.isCollapsed !== this.props.isCollapsed) {
+      updatedState.height = this.props.isCollapsed
+        ? 0
+        : this.contentNode.current.scrollHeight;
+    }
+
+    if (Object.keys(updatedState).length > 0) {
+      this.setState(updatedState);
+    }
+  }
+
+  /** Sums the 'items' and 'value' properties to form a group total.
+   * @param {array} values - The values to itemize.
+   * @returns {integer} - Returns the sum.
+   */
+  sumTotal(values) {
+    return values.reduce((acc, obj) => {
+      let accumulator = acc;
+      accumulator += obj.value || 0;
+      accumulator += obj.items
+        ? obj.items.reduce((a, o) => {
+            return 'value' in o ? o.value + a : a;
+          }, 0)
```

Comment:
```suggestion
          }, obj.value || 0)
```
Couldn't you simplify with doing something like this?

Reply:
This actually doesn't need to be here at all so I'm going to remove it, I borrowed some of this function from the other itemization component, and `items` doesn't exist as a concept here.